### PR TITLE
[candi] still bashible-api without bundle

### PR DESCRIPTION
**Pull request by sprait**
## Description

Preparatory phase for bashible without bundles:

1. Delete `node-group` context in bashible-api, save only `all` and `cluster-bootstrap` contexts.
2. Delete runtype `image_building` in bashible steps.
3. Resource=bootstrap (bashible.deckhouse.io/v1alpha1) without reference to bundle.
4. Resource=nodegroupbundles (bashible.deckhouse.io/v1alpha1) always return one bundle with a difference only in NGC (if any).
5. Move `AllowedBundles` from moduleconfig values to internalValues(in next iteration we will fully remove this param). 

## Why do we need it, and what problem does it solve?

Bashible optimization. The way to distro-independence. 

## What is the expected result?

<details>
  <summary>3. Resource=bootstrap (bashible.deckhouse.io/v1alpha1) without reference to bundle.</summary>

```
[root@dev-master-0 ~]# kubectl get bootstrap.bashible.deckhouse.io master -o json
{
    "apiVersion": "bashible.deckhouse.io/v1alpha1",
    "bootstrap": "#!/usr/bin/env bash\n\nfunction detect_bundle() {\n  #!/bin/bash\n  \n  # Copyright 2022 Flant JSC\n  # Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE\n  \n  function name_is_not_supported() {\n      \u003e\u00262 echo \"ERROR: ${PRETTY_NAME} is not supported.\"\n      exit 1\n  }\n  \n  function try_bundle(){\n    \u003e\u00262 echo \"WARNING: Trying to use ${1} bundle as default for: ${PRETTY_NAME}\"\n    echo \"${1}\"\n    exit 0\n  }\n  \n  if [ ! -e /etc/os-release ]; then\n    \u003e\u00262 echo \"ERROR: Can't determine OS! /etc/os-release is not found.\"\n    exit 1\n  fi\n  \n  . /etc/os-release\n  case \"$ID\" in\n    centos|rocky|almalinux|rhel)\n      case \"$VERSION_ID\" in 7|7.*|8|8.*|9|9.*)\n        echo \"centos\" \u0026\u0026 exit 0 ;;\n      esac\n      name_is_not_supported\n    ;;\n    redos)\n      case \"$VERSION_ID\" in 7|7.*|8|8.*)\n        echo \"redos\" \u0026\u0026 exit 0 ;;\n      esac\n      name_is_not_supported\n    ;;\n    rels|rosa)\n      case \"$VERSION_ID\" in 7.9)\n          echo \"rosa\" \u0026\u0026 exit 0 ;;\n      esac\n      case \"$VERSION\" in 12.4|12.5.*)\n          echo \"rosa\" \u0026\u0026 exit 0 ;;\n      esac\n      name_is_not_supported\n    ;;\n    ubuntu)\n      case \"$VERSION_ID\" in 18.04|20.04|22.04|24.04)\n        echo \"ubuntu-lts\" \u0026\u0026 exit 0 ;;\n      esac\n      name_is_not_supported\n    ;;\n    debian)\n      case \"$VERSION_ID\" in 10|11|12)\n        echo \"debian\" \u0026\u0026 exit 0 ;;\n      esac\n      name_is_not_supported\n    ;;\n    astra)\n      case \"$VERSION_ID\" in\n        1.7|1.7*|1.8|1.8*)\n          echo \"astra\" \u0026\u0026 exit 0 ;;\n        2.12|2.12.*)\n          echo \"debian\" \u0026\u0026 exit 0 ;;\n      esac\n      name_is_not_supported\n    ;;\n    altlinux)\n      case \"$VERSION_ID\" in p10|10|10.0|10.1|10.2|11)\n        echo \"altlinux\" \u0026\u0026 exit 0 ;;\n      esac\n      name_is_not_supported\n    ;;\n    \"\")\n      \u003e\u00262 echo \"ERROR: Can't determine OS! No ID in /etc/os-release.\"\n      exit 1\n    ;;\n  esac\n  \n  # try to determine os by ID_LIKE\n  for ID in $ID_LIKE; do\n    case \"$ID\" in\n      centos|rhel)\n        try_bundle \"centos\"\n      ;;\n      debian)\n        try_bundle \"debian\"\n      ;;\n      altlinux)\n        try_bundle \"altlinux\"\n      ;;\n    esac\n  done\n  \n  # try to determine os by packet manager\n  bundle=\"debian\"\n  if yum -q --version \u003e/dev/null 2\u003e/dev/null; then\n    bundle=\"centos\"\n  fi\n  try_bundle \"${bundle}\"\n  \n}\n\nfunction get_bundle() {\n  resource=\"$1\"\n  name=\"$2\"\n  token=\"$(\u003c/var/lib/bashible/bootstrap-token)\"\n\n  while true; do\n    for server in 192.168.199.84:6443; do\n      url=\"https://$server/apis/bashible.deckhouse.io/v1alpha1/${resource}s/${name}\"\n      if d8-curl -sS -f -x \"\" -X GET \"$url\" --header \"Authorization: Bearer $token\" --cacert \"$BOOTSTRAP_DIR/ca.crt\"\n      then\n       return 0\n      else\n        \u003e\u00262 echo \"failed to get $resource $name with curl https://$server...\"\n      fi\n    done\n    sleep 10\n  done\n}\n\nset -Eeuo pipefail\nshopt -s failglob\n\nexport BOOTSTRAP_DIR=\"/var/lib/bashible\"\nexport TMPDIR=\"/opt/deckhouse/tmp\"\nmkdir -p \"$BOOTSTRAP_DIR\" \"$TMPDIR\"\n\n# Directory contains sensitive information\nchmod 0700 $BOOTSTRAP_DIR\n\n# Detect bundle\nBUNDLE=\"$(detect_bundle)\"\n\n# set proxy env variables\nunset HTTP_PROXY http_proxy HTTPS_PROXY https_proxy NO_PROXY no_proxy\nexport D8_NODE_HOSTNAME=$(hostname -s)\n\nexport PATH=\"/opt/deckhouse/bin:$PATH\"\n# Get bashible script from secret\nget_bundle bashible \"${BUNDLE}.master\" | jq -r '.data.\"bashible.sh\"' \u003e $BOOTSTRAP_DIR/bashible.sh\nchmod +x $BOOTSTRAP_DIR/bashible.sh\n\n# Bashible first run\nuntil /var/lib/bashible/bashible.sh; do\n  echo \"Error running bashible script. Retry in 10 seconds.\"\n  sleep 10\ndone\n",
    "kind": "Bootstrap",
    "metadata": {
        "creationTimestamp": "2024-09-20T14:02:25Z",
        "name": "master"
    }
}
```

</details>

<details>
  <summary>4. Resource=nodegroupbundles (bashible.deckhouse.io/v1alpha1) always return one bundle </summary>
Check count of steps in different nodegroupbundles:

```
[root@dev-master-0 ~]# kubectl get nodegroupbundles.bashible.deckhouse.io ubuntu-lts.worker -o json  | jq -r '.data | length'
64
[root@dev-master-0 ~]#  kubectl get nodegroupbundles.bashible.deckhouse.io debian.worker -o json  | jq -r '.data | length'
64
[root@dev-master-0 ~]# kubectl get nodegroupbundles.bashible.deckhouse.io centos.worker -o json  | jq -r '.data | length'
64
```

Create NGCs that work in `ubuntu-lts` bundle and recheck the number of steps in nodegroupbundles:

```
[root@dev-master-0 ~]# cat <<EOF | kubectl apply -f -
apiVersion: deckhouse.io/v1alpha1
kind: NodeGroupConfiguration
metadata:
  name: ngc-only-ubuntu-lts
spec:
  bundles:
  - ubuntu-lts
  content: |
    exit 0
  nodeGroups:
  - '*'
  weight: 10
EOF

nodegroupconfiguration.deckhouse.io/ngc-only-ubuntu-lts created
[root@dev-master-0 ~]# kubectl get nodegroupbundles.bashible.deckhouse.io ubuntu-lts.worker -o json  | jq -r '.data | length'
65
[root@dev-master-0 ~]#  kubectl get nodegroupbundles.bashible.deckhouse.io debian.worker -o json  | jq -r '.data | length'
64
[root@dev-master-0 ~]# kubectl get nodegroupbundles.bashible.deckhouse.io centos.worker -o json  | jq -r '.data | length'
64
```
Show diff between nodegroupbundles:
```
# diff /tmp/ubuntu-lts.worker /tmp/centos.worker
34d33
<   "010_ngc-only-ubuntu-lts": "exit 0\n",
```
</details>

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi, dhctl, node-manager
type: feature
summary: Preparatory phase for bashible without bundles.
impact_level: default
```

### CODE REVIEW

File: candi/bashible/common-steps/all/000_fix_path_in_root_bashrc.sh.tpl [hunk 1]
```diff
@@ -12,6 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-if grep -q "^PATH=" /root/.bashrc 2>/dev/null; then
-  sed 's/^PATH=.*//' -i /root/.bashrc
-fi
+# if grep -q "^PATH=" /root/.bashrc 2>/dev/null; then
+#  sed 's/^PATH=.*//' -i /root/.bashrc
+# fi
```
Comment (sprait) [inline candi/bashible/common-steps/all/000_fix_path_in_root_bashrc.sh.tpl:17]: .

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/aws'...

**Comment by sprait** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 10943363968 10943363968-1-con-1-27 /sys/deckhouse-oss/install:pr9761 ec2-user@3.66.153.192

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by sprait** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 10943402987 10943402987-1-con-1-27 /sys/deckhouse-oss/install:pr9761 ec2-user@3.124.53.75

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/aws'...

**Comment by sprait** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 10960854727 10960854727-1-con-1-27 /sys/deckhouse-oss/install:pr9761 ec2-user@3.71.238.172

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/vcd'...

File: ee/be/candi/bashible/bundles/altlinux/all/000_fix_path_in_root_bashrc.sh.tpl [hunk 2]
```diff
@@ -1,17 +0,0 @@
...
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-if grep -q "^PATH=" /root/.bashrc 2>/dev/null; then
-  sed 's/^PATH=.*//' -i /root/.bashrc
-fi
```
Comment (sprait) [inline ee/be/candi/bashible/bundles/altlinux/all/000_fix_path_in_root_bashrc.sh.tpl:17]: this step looks strange, is it still needed now?

**Comment by sprait** [comment]
/e2e/abort vcd;Standard;containerd;1.27 10990850011 10990850011-1-con-1-27 /sys/deckhouse-oss/install:pr9761 ubuntu@192.168.254.63

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/vcd'...

File: ee/be/candi/bashible/bundles/altlinux/all/000_fix_path_in_root_bashrc.sh.tpl (hunk 2 above)
Comment (sprait) [reply ee/be/candi/bashible/bundles/altlinux/all/000_fix_path_in_root_bashrc.sh.tpl:17]: We discussed at the review that this step seems to be unnecessary.

**Comment by sprait** [comment]
/e2e/abort vcd;Standard;containerd;1.27 10991910456 10991910456-1-con-1-27 /sys/deckhouse-oss/install:pr9761 ubuntu@192.168.254.65

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/vcd'...

**Comment by sprait** [comment]
/e2e/abort vcd;Standard;containerd;1.27 10992551612 10992551612-1-con-1-27 /sys/deckhouse-oss/install:pr9761 ubuntu@192.168.254.68

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've recognized your '/e2e/abort' command and started the workflow...

**Review by z9r5 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/vcd'...

File: candi/bashible/common-steps/all/002_apt_timeout.sh.tpl [hunk 3]
```diff
@@ -12,8 +12,16 @@
...
+  if [[ -f /etc/apt/apt.conf.d/99timeout ]]; then
+    return 0
+  fi
+
+  echo 'Acquire::http::Timeout "120";' > /etc/apt/apt.conf.d/99timeout
+}
+
+case $(bb-is-bundle) in
+  debian|ubuntu-lts|astra|altlinux) set_apt_timeout ;;
+esac
```
Comment (RomanenkoDenys) [inline candi/bashible/common-steps/all/002_apt_timeout.sh.tpl:25]: May be simplify this step ?
Something like
if [[ -d /etc/apt/apt.conf.d/ ]]; then do work

File: candi/cloud-providers/aws/bashible/common-steps/all/000_install_nvme_cli.sh.tpl [hunk 4]
```diff
@@ -12,4 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-bb-yum-install nvme-cli
+case $(bb-is-bundle) in
+  debian|ubuntu-lts) bb-apt-install nvme-cli ;;
+  centos) bb-yum-install nvme-cli ;;
+esac
```
Comment (RomanenkoDenys) [inline candi/cloud-providers/aws/bashible/common-steps/all/000_install_nvme_cli.sh.tpl:18]: Can we self-compile static nvme-cli ?

File: candi/bashible/common-steps/all/002_apt_timeout.sh.tpl (hunk 3 above)
Comment (sprait) [reply candi/bashible/common-steps/all/002_apt_timeout.sh.tpl]: done

File: candi/cloud-providers/aws/bashible/common-steps/all/000_install_nvme_cli.sh.tpl (hunk 4 above)
Comment (sprait) [reply candi/cloud-providers/aws/bashible/common-steps/all/000_install_nvme_cli.sh.tpl:18]: nope, static build didn't work for this package.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/vcd'...

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/static'...

**Review by z9r5 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/aws'...

**Comment by sprait** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 12118918771 12118918771-1-con-1-27 /sys/deckhouse-oss/install:pr9761 ec2-user@18.153.164.192

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @sprait. I've started the workflow for label 'e2e/run/aws'...

**Pull request merged**